schema/error.go

```diff
@@ -0,0 +1,79 @@
+package schema
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+)
+
+// ErrorMap contains a map of errors by field name.
+type ErrorMap map[string][]interface{}
+
+// Error implements the built-in error interface.
+func (e ErrorMap) Error() string {
+	errs := make([]string, 0, len(e))
+	for key := range e {
+		errs = append(errs, key)
+	}
+	sort.Strings(errs)
+	for i, key := range errs {
+		errs[i] = fmt.Sprintf("%s is %s", key, e[key])
+	}
+	return strings.Join(errs, ", ")
+}
+
+// Extend copies all errors from err to e.
+func (e ErrorMap) Extend(err ErrorMap) {
+	for k, v := range err {
+		e[k] = append(e[k], v...)
+	}
+}
+
+// Tidy returns an un-typed nil if e is empty, and an ErrorMap otherwise.
+func (e ErrorMap) Tidy() error {
```

Comment:
That's a strange API. Why do you need that? Couldn't we just expose a `Len` method and let the caller do what's appropriate?

Reply:
I agree, it's strange. Will change it. Len could work.

Reply:
Fixed